packages/@n8n/ai-workflow-builder.ee/src/tools/utils/node-configuration.utils.ts

```diff
@@ -0,0 +1,146 @@
+import type { INodeParameters } from 'n8n-workflow';
+
+import { MAX_NODE_EXAMPLE_CHARS } from '@/constants';
+import type { NodeConfigurationsMap, NodeConfigurationEntry, WorkflowMetadata } from '@/types';
+
+/**
+ * Node structure for configuration collection
+ */
+interface NodeForConfiguration {
+	type: string;
+	typeVersion: number;
+	parameters: INodeParameters;
+}
+
+/**
+ * Collect configuration from a single node if it meets size requirements
+ * Returns null if the node has no parameters or exceeds size limits
+ */
+export function collectSingleNodeConfiguration(
+	node: NodeForConfiguration,
+): NodeConfigurationEntry | null {
+	const hasParams = Object.keys(node.parameters).length > 0;
+	if (!hasParams) return null;
+
+	const parametersStr = JSON.stringify(node.parameters);
+	if (parametersStr.length > MAX_NODE_EXAMPLE_CHARS) return null;
+
+	return {
+		version: node.typeVersion,
+		parameters: node.parameters,
+	};
+}
+
+/**
+ * Add a node configuration to a configurations map
+ * Mutates the map in place for efficiency when processing multiple nodes
+ */
+export function addNodeConfigurationToMap(
+	nodeType: string,
+	config: NodeConfigurationEntry,
+	configurationsMap: NodeConfigurationsMap,
+): void {
+	if (!configurationsMap[nodeType]) {
+		configurationsMap[nodeType] = [];
+	}
+	configurationsMap[nodeType].push(config);
+}
+
+/**
+ * Collect node configurations from multiple workflows
+ * Does not generate mermaid diagrams - just extracts node configurations
+ */
+export function collectNodeConfigurationsFromWorkflows(
+	workflows: WorkflowMetadata[],
+): NodeConfigurationsMap {
+	const configurations: NodeConfigurationsMap = {};
+
+	for (const workflow of workflows) {
+		for (const node of workflow.workflow.nodes) {
+			// Skip sticky notes
+			if (node.type === 'n8n-nodes-base.stickyNote') continue;
```

Comment:
could also filter out custom node types or unknown node type to an instance. Maybe that's not important if we never request those nodes